contracts/src/TeleporterTokenDestination.sol

```diff
@@ -104,17 +104,6 @@ abstract contract TeleporterTokenDestination is
             );
         }
 
-        // Deposit the funds sent from the user to the bridge,
-        // and set to adjusted amount after deposit
-        amount = _deposit(amount);
-        require(
-            amount > input.primaryFee + input.secondaryFee,
-            "TeleporterTokenDestination: insufficient amount to cover fees"
-        );
-
-        amount -= input.primaryFee;
-        _burn(amount);
-
```

Comment:
This was moved to be handled by the respective destination contracts because the multiplier on the native token destination needs special handling.

Reply:
It looks like the special handling required is not because of the multiplier but rather because `_deposit` should only deposit the fee amount in the `NativeTokenDestination` implementation, right?

IIUC, one alternative (not suggesting it) could for the `NativeTokenDestination.send` implementation would be to still "deposit" the full `msg.value` amount (converting it to WETH/WAVAX), and then changing the `_burn` implementation to transfer that ERC20 to the burn address (rather than the native token). Is that understanding correct?

Reply:
The special handling **_is_** required because of the multiplier. The fees have to be deducted before multiplying/dividing, and then the remaining tokens have to be scaled, which means the shared function couldn't handle it. 

`_deposit` only taking the fee amount was just a simplification I made on the other implementation. We would definitely deposit the full amount and then burn the ERC20, but that has the downside of extra native tokens being locked in the ERC20, and then we'd also have a roundabout way we'd have to compute the total supply for the chain.

Reply:
I also think we should look at depositing the full `amount` then subtracting primary fee from it for burning, because as [this](https://github.com/ava-labs/teleporter-token-bridge/pull/49/files#diff-72a6aff99c287bf6418f12e1520a21daf9ea6b682bd985a8645be84e6303c990R185) points out, we'll need to deposit full amount to figure out actual amount deposited, and make sure full primary fee amount is deposited. 

Then this leaves us with the same structure as previously, where the logic stays in `TeleporterTokenDestination`. The question is then for `NativeTokenDestination` has to scale the amount being encoded in the message for `sendTeleporterMessage`. 

- We could add a `beforeTokenTransfer` or `scaleTokens` call that needs to be implemented by `TeleporterTokenDestination`s before the `sendTeleporterMessage`. This might not be as applicable to our default `ERC20Destination` implementation, but can be useful for other implementations people come up with.
- Would `scaleTokens` be something we want to bring to `ERC20Destination`?

Reply:
Another option is an `_encodeMessage` function that encodes the message in `TeleporterMessageInput` for calling `_sendTeleporterMessage`. This way `NatievTokenDestination` can encode its message with the scaled amount. And also, in `reportTotalBurnedFees` can call `_encodeMessage` instead of copying the logic on forming `SendTokensInput`

Reply:
Ohhh hmm... I think I like the `_encodeMessage` option? One odd issue I ran into with `_send` is that it requires `calldata` to be passed in, and I wasn't able to create `calldata` in the function, only `memory`.